mmf/models/transformers/base.py

```diff
@@ -134,29 +122,48 @@ def get_optimizer_parameters(self, config):
                         parameters=parameters,
                         param_list=param_list,
                     )
-            else:
+            elif name != "heads":
                 # For other modules in trunk, add to same param group
                 param_list += list(module.named_parameters())
-
+                backbone_param_set.update(list(module.parameters()))
+        head_configs = self.config.get("heads", [])
+        # Heads can have different learning rates. This is handled here
+        if len(head_configs) > 0:
+            # Parameters in the head which have a separate learning
+            # rate, are added as a separate param group
+            for head_config, head in zip(head_configs, self.heads):
+                parameters, param_list = self.set_lr_for_parameters(
+                    config=head_config,
+                    module_name="{} head".format(head_config.get("type", "MLP")),
+                    base_lr=lr,
+                    module=head,
+                    parameters=parameters,
+                    param_list=param_list,
+                    backbone_param_set = backbone_param_set
+                )
         parameters += get_bert_configured_parameters(param_list)
 
         return parameters
 
     def set_lr_for_parameters(
-        self, config, module_name, base_lr, module, parameters, param_list
+        self, config, module_name, base_lr, module, parameters, param_list, backbone_param_set = None
     ):
         lr_multiplier = config.get("lr_multiplier", 1.0)
+        if backbone_param_set is None:
+            module_param = list(module.named_parameters())
+        else:
```

Comment:
Now, you can remove this else condition.

Reply:
Thanks for your reviewing. I will change my code according to ur suggestion.